Feat: Add git branches to Algernon

leonardhorns opened the pull request.

leonardhorns committed:
Refactor: Clean up and extend ALGTestGitProjectType

- rename test class to ALGTestGitTypes
- refactor setUpTestWorkingCopy
- recategorize git project test methods
- match #expectedFailures to new categorization
- add branches to mockWorkingCopy

leonardhorns committed:
Test: Add tests for ALGGitBranchType

- test #getGitBrowser has proper branch selected
- test action open
- test appearance in default search and as children
- test has-children

leonardhorns committed:
Feat: Add git branch type to Algernon

- add subclasses ALGGitRemoteBranchType, ALGGitLocalBranchType
- add actions open, switch
- add git branches to git project type as children
- add keyword 'switch'
- extend ALGLinting class exceptions to git branch types
- (current icons are temporary placeholders)

Resolves #266

leonardhorns committed:
Test: Add tests for git branch type switch-action

- refactor ALGFakeActionTarget
- add ALGFakeBranch
- add tests

yannik-dittmar commented:
PR is still a draft, not reviewing. Also the test coverage seems to have decreased quite a bit. Have you written test for your stuff yet?

leonardhorns commented:
Tests are done, but are currently still marked as expected failures until `SquitBrowser` is added to the CI on this branch. That should resolve the coverage issue.
Review can wait until marked ready for review...

leonardhorns committed:
Fix: Declare global variable 'ALGRemoteBranch' before running ALGLinting

leonardhorns committed:
Test-Fix: Set up GitStamp before running tests for git types

- set values for 'DefaultName' and 'DefaultEmail'
- restore previous values during #tearDown

gwauge commented on packages/AlgernonTests.package/ALGLinting.class/instance/collectTestObjects.st:
```diff
@@ -4,5 +4,10 @@ collectTestObjects
+			  ALGGitBranchType.
+			  ALGGitLocalBranchType.
+			  ALGGitRemoteBranchType}].
 	^ classes collect: [ :each | each as: SLTestObject ].
```
Punkt am Ende weg

gwauge commented on packages/AlgernonTests.package/ALGTestGitTypes.class/instance/setUpGitStamp.st:
```diff
@@ -0,0 +1,10 @@
+		previousGitStampDefaultEmail: ([GitStamp defaultEmail] on: GitStampEmailRequest do: ['']).
+	
+	GitStamp
+		defaultName: 'Hasso';
```
❤️

gwauge reviewed (approved):
Sau starke Arbeit Leo!

Hab nur den einen Punkt gesehen wegen coding standards aber den kannst du eigentlich auch lassen sonst und wir refactoren ihn dann auf dem dev.
